### ISSUE TITLE
Minions - Repeated gettimeofday, clock_gettime, and poll calls

### ISSUE THREAD

**Issue by knagra** [saltstack/salt#19975]
Not sure if this is a related bug or the same bug as https://github.com/saltstack/salt/issues/18593, but I'm getting the same problem with minions. There seems to be a `gettimeofday` call about every second. Here's an `strace` readout:
```
[...]
[knagra@dtdstesttwo01 ~]$ sudo strace -p 10930
Process 10930 attached - interrupt to quit
restart_syscall(<... resuming interrupted call ...>) = 0
poll([{fd=32, events=POLLIN}], 1, 0)    = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {5753791, 348396954}) = 0
gettimeofday({1421962578, 912296}, NULL) = 0
clock_gettime(CLOCK_MONOTONIC, {5753791, 348660511}) = 0
poll([{fd=38, events=POLLIN}], 1, 0)    = 0 (Timeout)
poll([{fd=38, events=POLLIN}], 1, 0)    = 0 (Timeout)
poll([{fd=38, events=POLLIN}], 1, 1000) = 0 (Timeout)
poll([{fd=38, events=POLLIN}], 1, 0)    = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {5753792, 349882042}) = 0
gettimeofday({1421962579, 913820}, NULL) = 0
gettimeofday({1421962579, 913934}, NULL) = 0
clock_gettime(CLOCK_MONOTONIC, {5753792, 350268714}) = 0
poll([{fd=10, events=POLLIN}], 1, 0)    = 0 (Timeout)
poll([{fd=10, events=POLLIN}], 1, 0)    = 0 (Timeout)
poll([{fd=10, events=POLLIN}], 1, 1)    = 0 (Timeout)
poll([{fd=10, events=POLLIN}], 1, 0)    = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {5753792, 351539699}) = 0
gettimeofday({1421962579, 915427}, NULL) = 0
clock_gettime(CLOCK_MONOTONIC, {5753792, 351807947}) = 0
poll([{fd=32, events=POLLIN}], 1, 0)    = 0 (Timeout)
poll([{fd=32, events=POLLIN}], 1, 0)    = 0 (Timeout)
poll([{fd=32, events=POLLIN}], 1, 1000) = 0 (Timeout)
poll([{fd=32, events=POLLIN}], 1, 0)    = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {5753793, 353058999}) = 0
gettimeofday({1421962580, 916984}, NULL) = 0
clock_gettime(CLOCK_MONOTONIC, {5753793, 353353288}) = 0
poll([{fd=38, events=POLLIN}], 1, 0)    = 0 (Timeout)
poll([{fd=38, events=POLLIN}], 1, 0)    = 0 (Timeout)
poll([{fd=38, events=POLLIN}], 1, 1000) = 0 (Timeout)
poll([{fd=38, events=POLLIN}], 1, 0)    = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {5753794, 354569172}) = 0
gettimeofday({1421962581, 918430}, NULL) = 0
gettimeofday({1421962581, 918482}, NULL) = 0
[...]
```
CPU was idle 99% of the time, unlike the salt-master issue in #18593. We have over 700 minions in a multi-master setup; so I can't really check each one, but every one that I checked had the same repeated `gettimeofday`, `clock_gettime`, and `poll` calls.

Here are file descriptors for the salt-minion process:
```
[knagra@dtdstesttwo01 ~]$ sudo ls -l /proc/10930/fd/
total 0
lrwx------ 1 root root 64 Jan 22 13:42 0 -> /dev/null
lrwx------ 1 root root 64 Jan 22 13:42 1 -> /dev/null
lrwx------ 1 root root 64 Jan 22 13:42 10 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 11 -> socket:[47740473]
lrwx------ 1 root root 64 Jan 22 13:42 12 -> socket:[47740475]
lrwx------ 1 root root 64 Jan 22 13:42 13 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 14 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 15 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 16 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 17 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 18 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 19 -> socket:[47743788]
lrwx------ 1 root root 64 Jan 22 13:42 2 -> /dev/null
lrwx------ 1 root root 64 Jan 22 13:42 20 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 21 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 22 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 23 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 24 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 25 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 26 -> socket:[47747871]
lrwx------ 1 root root 64 Jan 22 13:42 27 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 28 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 29 -> anon_inode:[eventpoll]
l-wx------ 1 root root 64 Jan 22 13:42 3 -> /var/log/salt/minion
lrwx------ 1 root root 64 Jan 22 13:42 30 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 31 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 32 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 33 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 34 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 35 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 36 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 37 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 38 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 39 -> socket:[47747872]
lrwx------ 1 root root 64 Jan 22 13:42 4 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 40 -> socket:[47743789]
lrwx------ 1 root root 64 Jan 22 13:42 5 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 6 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 7 -> anon_inode:[eventfd]
lrwx------ 1 root root 64 Jan 22 13:42 8 -> anon_inode:[eventpoll]
lrwx------ 1 root root 64 Jan 22 13:42 9 -> anon_inode:[eventfd]
```
So the `poll` calls are all to anon_inodes (10, 32, 38). Are those event trigger files for incoming connections?

We've been getting really slow responses from minions (most of them fail to respond in the 15 second time limit), and I ran across this issue as a possible cause of our problem.

**Comment by knagra** [comment]
Here's a versions report from a master:
```
[knagra@dsalt01 ~]$ salt --versions-report
           Salt: 2014.7.0
         Python: 2.6.6 (r266:84292, Jan 22 2014, 09:42:36)
         Jinja2: 2.7.2
       M2Crypto: 0.20.2
 msgpack-python: 0.4.0
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 2.2.0.1
           RAET: Not Installed
            ZMQ: 3.2.4
           Mako: Not Installed
```

**Comment by knagra** [comment]
And from a minion:
```
[knagra@dtdstesttwo01 ~]$ salt-minion --versions-report
           Salt: 2014.7.0
         Python: 2.6.6 (r266:84292, Feb 22 2013, 00:00:18)
         Jinja2: 2.7.2
       M2Crypto: 0.20.2
 msgpack-python: 0.4.0
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 2.2.0.1
           RAET: Not Installed
            ZMQ: 3.2.4
           Mako: Not Installed
```
I think all the minions are running the same versions of everything.

**Comment by rallytime** [comment]
Thanks for the report @knagra. It does look similar, but I'm not sure if it is exactly the same. Are you using the salt scheduler?

**Comment by knagra** [comment]
Ah. I changed the minion config for one of the minion boxes from:
```
loop_interval: 1
```
to:
```
loop_interval: 60
```
which seems to have resolved the repeated calls to all three functions. They're happening only once a minute now.

The 1 -> 60 change actually exacerbates the no response problem from that minion (even longer delay in response), but I'll open a separate ticket when more symptoms manifest.

Thanks for the lead, @rallytime.

**Comment by rallytime** [comment]
Yeah, sure thing! Is there anything else that should be fixed here, or is this particular issue resolved by changing the loop interval?

**Comment by knagra** [comment]
Yeah, I suppose it's technically a non-issue. That's expected and correct behavior it seems.

**Comment by rallytime** [comment]
Great! Thanks!
